src/taskpane/server.js

```diff
@@ -1135,40 +1335,70 @@ function contentValidationSetter(sheet, model, fieldTypeEnums, context) {
         break;
     }
   }
+  if (!IS_GOOGLE) {
+    return context.sync();
+  }
 }
 
 
-
 // create dropdown validation on set column based on specified values
 // @param: sheet - the sheet object
 // @param: columnNumber - int of the column to validate
 // @param: rowLength - int of the number of rows for range (global param)
 // @param: list - array of values to show in a dropdown and use for validation
 // @param: allowInvalid - bool to state whether to restrict any values to those in validation or not
-function setDropdownValidation(sheet, columnNumber, rowLength, list, allowInvalid) {
+async function setDropdownValidation(mainSheetName, columnNumber, list, allowInvalid, context, model) {
   if (IS_GOOGLE) {
-    // create range
-    var range = sheet.getRange(2, columnNumber, rowLength);
+    //first, write the values to the dbSheet
+    var dataBaseSheetName = params.dataSheetName + model.currentProject.id + model.currentArtifact.id;
+    mainSheetName = model.currentArtifact.name + ", PR-" + model.currentProject.id;
+
+    var values = [];
+    list.forEach(function (item) {
+      var itemArray = [item];
+      values.push(itemArray);
+    });
+
+    var dbSheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName(dataBaseSheetName),
+      dbRange = dbSheet.getRange(1, columnNumber, list.length, 1);
+    dbRange.setValues(values);
 
+    //Now, point the fields in the mainsheet to the database worksheet (source)
+    var sheet = SpreadsheetApp.getActiveSpreadsheet().getSheetByName(mainSheetName);
+    var range = sheet.getRange(2, columnNumber, sheet.getMaxRows() - 1);
     // create the validation rule
     // requireValueInList - params are the array to use, and whether to create a dropdown list
     var rule = SpreadsheetApp.newDataValidation()
-      .requireValueInList(list, true)
+      .requireValueInRange(dbRange, true)
       .setAllowInvalid(allowInvalid)
       .build();
     range.setDataValidation(rule);
-
   } else {
-    var range = sheet.getRangeByIndexes(1, columnNumber - 1, rowLength, 1);
+    //max rows for Excel
+    var nonHeaderRows = 1048576 - 1;

```

Comment:
Is 1048576 the number of rows in Excel always? Consider changing this to a const like `const EXCEL_NUMBER_OF_ROWS = 1048576;`